src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2587,32 +2566,41 @@ public override BoundNode VisitConditionalOperator(BoundConditionalOperator node
             {
                 resultType = node.Type.SetUnknownNullabilityForReferenceTypes();
                 resultState = NullableFlowState.NotNull;
-            }
-            else
-            {
-                var resultTypeWithAnnotations = TypeWithAnnotations.Create(resultType);
-                TypeWithState convertedConsequenceResult = default;
-                TypeWithState convertedAlternativeResult = default;
 
-                if (consequenceEndReachable)
+                var resultTypeWithState = TypeWithState.Create(resultType, resultState);
+                var visitResult = new VisitResult(resultTypeWithState, resultTypeWithState.ToTypeWithAnnotations());
+
+                if (consequence != node.Consequence)
                 {
-                    convertedConsequenceResult = convertResult(
-                        node.Consequence,
-                        consequence,
-                        consequenceConversion,
-                        resultTypeWithAnnotations,
-                        consequenceRValue);
+                    Debug.Assert(node.Consequence.Kind == BoundKind.Conversion);
+                    SetAnalyzedNullability(node.Consequence, visitResult);
                 }
 
-                if (alternativeEndReachable)
+                if (alternative != node.Alternative)
                 {
-                    convertedAlternativeResult = convertResult(
-                        node.Alternative,
-                        alternative,
-                        alternativeConversion,
-                        resultTypeWithAnnotations,
-                        alternativeRValue);
+                    Debug.Assert(node.Alternative.Kind == BoundKind.Conversion);
+                    SetAnalyzedNullability(node.Alternative, visitResult);
                 }
```

Comment:
>} [](start = 16, length = 1)

Do you have some tests showing the effect of this change?

Reply:
I ended up not being able to find a case where consequence != node.Consequence when there's no result type, so I added an assertion instead and removed this code.

---
In reply to: [282221747](https://github.com/dotnet/roslyn/pull/35544#discussion_r282221747) [](ancestors = 282221747)